malaria-app-android/src/main/java/com/peacecorps/malaria/ui/user_profile/UserProfileFragment.java

```diff
@@ -0,0 +1,193 @@
+
+public class UserProfileFragment extends BaseFragment implements UserProfileContract.UserProfileMvpView {
+
+    private OnUserFragmentListener mListener;
+    private Context context;
+    //Butterknife variables
+    @BindView(R.id.edit_text_user_name)
+    EditText userName;
+    @BindView(R.id.edit_text_user_email)
+    EditText userEmail;
+    @BindView(R.id.edit_text_user_age)
+    EditText userAge;
+    @BindView(R.id.edit_text_user_medicine_type)
+    EditText userMedicationType;
+
+    private UserProfilePresenter<UserProfileFragment> presenter;
+
+    public static UserProfileFragment newInstance() {
+//        Bundle args = new Bundle();
+//        args.putString(ARG_PARAM1, param1);
+//        args.putString(ARG_PARAM2, param2);
+//        fragment.setArguments(args);
+        return new UserProfileFragment();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view =  inflater.inflate(R.layout.user_profile, container, false);
+        context = getContext();
+        if(getActivity()!=null)
+            ButterKnife.bind(this, view);
+        presenter = new UserProfilePresenter<>(InjectionClass.provideDataManager(context), context);
+        presenter.attachView(this);
+        return view;
+    }
+
+    @Override
+    protected int getContentResource() {
+        return R.layout.user_profile;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        init();
+    }
+
+    @Override
+    protected void init() {
+        presenter.setPreviousDetails();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof OnUserFragmentListener) {
+            mListener = (OnUserFragmentListener) context;
+        } else {
+            throw new RuntimeException(context.toString()
+                    + " must implement OnFragmentInteractionListener");
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mListener = null;
+        presenter.detachView();
+        presenter = null;
+    }
+
+    // set initial details from shared preferences
+    @Override
+    public void setInitialValuesIfAvailable(String name, String email, int age, String medicine) {
+        if(!name.equals(""))
+            userName.setText(name);
+        if(!email.equals(""))
+            userEmail.setText(email);
+        if(age>0)
+            userAge.setText(String.valueOf(age));
+        userMedicationType.setText(medicine);
+    }
+
+    @Override
+    public String getUserName() {
+        if (userName == null) {
```

Comment:
what if userName is empty??

Reply:
userName is an edit text, how can it be empty? getUserName will return trimmed values, returned values is empty of not is checked in checkNameError() function before on Save button click.